src/Parser/FrostMage/Modules/Items/Tier20_2set.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage } from 'common/format';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+import Module from 'Parser/Core/Module';
+
+
+class Tier20_2set extends Module {
+  static dependencies = {
+    combatants: Combatants,
+  }
+
+  item() {
+    const T202setUptime = this.combatants.selected.getBuffUptime(SPELLS.FROST_MAGE_T20_2SET_BONUS_BUFF.id) / this.owner.fightDuration;
+    return {
+      id: `spell-${SPELLS.FROST_MAGE_T20_2SET_BONUS_BUFF.id}`,
+      icon: <SpellIcon id={SPELLS.FROST_MAGE_T20_2SET_BONUS_BUFF.id} />,
+      title: <SpellLink id={SPELLS.FROST_MAGE_T20_2SET_BONUS_BUFF.id} />,
```

Comment:
A few things here:

This should have an `on_initialized` that detects if player has the 2set and is only active then. Other SPELLS entries that have XXX_XXX_TXX_2SET_BONUS_BUFF are referring to a hidden aura that shows if the player has the set bonus. Those entries do NOT refer to the sets proc. The aura you're looking for for the frost mage 2t20 is this: http://www.wowhead.com/spell=242252/item-mage-t20-frost-2p-bonus. The 2t20 proc is called 'Frozen Mass', and should be named as such in its SPELLS entry.

Also, while you don't need to do this now, it is possible to calculate the actual percent damage increase you get from this bonus. That's a good candidate for future enhancement.

Reply:
Is there an existing spec that does this so i can potentially adopt that? I found examples of tier bonuses or talents that give a % damage increase using the getDamageBonus stuff .... but i didnt find any that give a boost to a secondary stat (like haste or in this case crit)

Reply:
Frozen Mass doesn't increase crit, it increases *crit damage*. It should be double checked, but I think this is just a +20% damage bonus that applies only to crits.

You can tell if a spell crit by checking the 'hitType' field in a damage event. A value of '2' indicates a crit (see https://github.com/WoWAnalyzer/WoWAnalyzer/blob/master/src/Parser/Core/HIT_TYPES.js). So for each damage event you'd check if the player has Frozen Mass and if the damage was a crit, and if so use getDamageBonus with a value of 0.20.